### ISSUE TITLE
col_default_is fails with timestampz(0)

### ISSUE THREAD

**Issue by kbrannen** [theory/pgtap#244]
This is with Pg 12.2 and pgtap 1.2.0.

Assuming a table like:
```
test=# \d device_software
                           Table "public.device_software"
    Column     |            Type             | Collation | Nullable |      Default
---------------+-----------------------------+-----------+----------+-------------------
 device_fk     | integer                     |           | not null |
 software      | character varying(16)       |           | not null |
 version       | character varying(16)       |           | not null |
 when_modified | timestamp(0) with time zone |           | not null | CURRENT_TIMESTAMP
```

SELECT col_type_is( 'device_software', 'when_modified', 'timestamp(0) with time zone' ); -- this works!
SELECT col_default_is( 'device_software', 'when_modified', 'CURRENT_TIMESTAMP' );        -- this fails
```
psql:device_software.pg:46: ERROR:  invalid input syntax for type timestamp with time zone: "CURRENT_TIMESTAMP"
LINE 1: ...s(CURRENT_TIMESTAMP::timestamp(0) with time zone, 'CURRENT_T...
                                                             ^
QUERY:  SELECT is(CURRENT_TIMESTAMP::timestamp(0) with time zone, 'CURRENT_TIMESTAMP'::timestamp(0) with time zone, 'Column device_software.when_modified should default to ''CURRENT_TIMESTAMP''')
CONTEXT:  PL/pgSQL function _def_is(text,text,anyelement,text) line 10 at EXECUTE
SQL statement "SELECT _def_is(
        pg_catalog.pg_get_expr(d.adbin, d.adrelid),
        pg_catalog.format_type(a.atttypid, a.atttypmod),
        $3, $4
    )
      FROM pg_catalog.pg_class c, pg_catalog.pg_attribute a, pg_catalog.pg_attrdef d
     WHERE c.oid = a.attrelid
       AND pg_table_is_visible(c.oid)
       AND a.atthasdef
       AND a.attrelid = d.adrelid
       AND a.attnum = d.adnum
       AND c.relname = $1
       AND a.attnum > 0
       AND NOT a.attisdropped
       AND a.attname = $2"
PL/pgSQL function _cdi(name,name,anyelement,text) line 13 at RETURN
Dubious, test returned 3 (wstat 768, 0x300)
```

If you change the col type in the table definition to "timestampz" without the precision indicator, then the message changes (and almost works but doesn't). So it's possible the problem is the way I wrote the 3rd arg, but the docs show that should work (and maybe this is a doc bug).

Speaking of which, you have a doc bug (1.2.0):
```
SELECT col_default_is( 'tab', age, NULL );
SELECT col_default_is( 'tab', age, NULL::integer );
```
On both of those lines, I believe "age" needs to be quoted.

**Comment by kbrannen** [comment]
Sorry, small correction to the above after checking again. If you make the column a "timestampz" in the table definition, then the col_default_is() call works just fine. So the precision indication is the source of the error from what I can figure out.

**Comment by kbrannen** [comment]
Just found a similar issue with CURRENT_DATE in another table, so maybe this is will all the CURRENT_* functions?

SELECT col_default_is('inventory','in_service','CURRENT_DATE');
```
psql:inventory.pg:43: ERROR:  invalid input syntax for type date: "CURRENT_DATE"
LINE 1: SELECT is(CURRENT_DATE::date, 'CURRENT_DATE'::date, 'Column ...
                                      ^
QUERY:  SELECT is(CURRENT_DATE::date, 'CURRENT_DATE'::date, 'Column inventory.in_service should default to ''CURRENT_DATE''')
...same context & sql as above...
```
It's interesting that the first 2 args of is() look the same.

OK, I think I get it now, changing to (no quotes):

SELECT col_default_is('inventory','in_service',CURRENT_DATE);

and it works with:

ok 10 - Column inventory.in_service should default to '2020-03-27'

That's really weird, but I like that it works.

So back to the original problem up above, changing the test to:

SELECT col_default_is( 'device_software', 'when_modified', CURRENT_TIMESTAMP );

gives:

ok 11 - Column device_software.when_modified should default to '2020-03-27 22:46:25.466848+00'

Therefore, this is really a doc bug after all. I'd followed the example where you show:

SELECT col_default_is( 'user', 'created_at', 'now()' );  -- from the doc

but I guess the CURRENT_* functions should NOT be quoted. I think that should be mentioned in the docs.

**Comment by theory** [comment]
I suspect the issue is the polymorphic functions in PostgreSQL. As the [`col_defautl_is()` docs](https://pgtap.org/documentation.html#col_default_is) say:

> The default argument must have an unambiguous type in order for the call to succeed. If you see an error such as ‘ERROR: could not determine polymorphic type because input has type “unknown”’, it’s because you forgot to cast the expected value, probably a NULL (which, by the way, you can only properly test for in PostgreSQL 8.3 and later), to its proper type. IOW, this will fail:
>
>    SELECT col_default_is( 'tab', age, NULL );
>
>But this will not:
>
>    SELECT col_default_is( 'tab', age, NULL::integer );

**Comment by theory** [comment]
Oh, but you're tripping on the `current_*` keywords, I see. I guess I've never used the special keyword functions before. Looking…

**Comment by theory** [comment]
You were right, there were some `CURRENT_*` keywords that weren't included. Should be fixed in 50c2585, thank you!

**Comment by kbrannen** [comment]
Thanks! :)
